### ISSUE TITLE
[Feature]: Video compression

### ISSUE THREAD

**Issue by pedrodelbosque** [simplex-chat/simplex-chat#3620]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Platform

all

### Feature

Sending a short video file, on phones with high-quality cameras, can quickly generate giant files, up to 200-300MB for a few seconds. Compressing videos before sending removes strain from servers and speeds up media transfers.

**Comment by epoberezkin** [comment]
yes, Android is more complex as it doesn't have it in available system libraries - we will need to bundle ffmpeg I think...

**Comment by pedrodelbosque** [comment]
Yes, bundling ffmpeg seems unavoidable, more so as in the future you would want to implement a basic video editor (cutting, cropping).
On a side note, compressing video will surely help performance-wise for the upcoming local video encryption.

**Comment by Rubber-Duckie** [comment]
recommend that this is optional behavour behind a feature toggle. 

sub setting: Disable/Enable transcoding (additional compression)
sub setting: Compression ratio (percent quality)
sub setting: Max size options [4k, 1080p, 720p, other common formats]
